ui/modal/modalPublishPreview/view.jsx

```diff
@@ -151,38 +152,48 @@ const ModalPublishPreview = (props: Props) => {
   const txFee = previewResponse ? previewResponse['total_fee'] : null;
   //   $FlowFixMe add outputs[0] etc to PublishResponse type
   const isOptimizeAvail = filePath && filePath !== '' && isVid && ffmpegStatus.available;
-  let modalTitle;
-  if (isStillEditing) {
-    if (livestream) {
-      modalTitle = __('Confirm Update');
-    } else {
-      modalTitle = __('Confirm Edit');
-    }
-  } else if (livestream) {
-    modalTitle = releasesInFuture ? __('Schedule Livestream') : __('Create Livestream');
-  } else {
-    modalTitle = __('Confirm Upload');
-  }
 
+  let modalTitle;
   let confirmBtnText;
-  if (!publishing) {
+  updateLabels();
+  function updateLabels() {
     if (isStillEditing) {
-      confirmBtnText = __('Save');
-    } else if (livestream) {
-      confirmBtnText = __('Create');
+      if (isLivestreamClaim) {
+        modalTitle = __('Confirm Update');
+      } else {
+        modalTitle = __('Confirm Edit');
+      }
+    } else if (isLivestreamClaim) {
+      modalTitle = releasesInFuture ? __('Schedule Livestream') : __('Create Livestream');
+    } else if (isMarkdownPost) {
+      modalTitle = __('Confirm Post');
     } else {
-      confirmBtnText = __('Upload');
+      modalTitle = __('Confirm Upload');
     }
-  } else {
-    if (isStillEditing) {
-      confirmBtnText = __('Saving');
-    } else if (livestream) {
-      confirmBtnText = __('Creating');
+
+    if (!publishing) {
+      if (isStillEditing) {
+        confirmBtnText = __('Save');
+      } else if (isLivestreamClaim) {
+        confirmBtnText = __('Create');
+      } else {
+        confirmBtnText = __('Upload');
+      }
     } else {
-      confirmBtnText = __('Uploading');
+      if (isStillEditing) {
+        confirmBtnText = __('Saving');
+      } else if (livestream) {
+        confirmBtnText = __('Creating');
+      } else {
+        confirmBtnText = __('Uploading');
+      }
     }
   }
 
+  React.useEffect(() => {
+    updateLabels();
+  }, [filePath, isMarkdownPost, isLivestreamClaim]);
```

Comment:
The dependency is missing `isStillEditing` and `publishing`, a common thing to miss when the effect calls another function.

If there is a real reason to ignore changes in those variables (i.e. something not obvious to the reader), add something like:
```
    // do NOT listen to messageValue change, otherwise will autofocus while typing
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [autoFocus, inputRef]);
```

Reply:
The entire function became redundant after removing the manual execution ;) The entire logic is now in the effect.